src/web/components/Search/SearchBar.tsx

```diff
@@ -6,24 +6,26 @@ import './SearchBar.scss';
 
 type SearchBarContainerProps = React.PropsWithChildren<{
   className?: string;
-  handleOnBlur: () => void;
+  handleOnBlur?: () => void;
 }>;
 export function SearchBarContainer({ children, className, handleOnBlur }: SearchBarContainerProps) {
   const componentRef = useRef<HTMLDivElement>(null);
-  const handleClick = (event: MouseEvent) => {
-    if (componentRef.current && !componentRef.current.contains(event.target as Node)) {
-      handleOnBlur();
-    }
-  };
 
   useEffect(() => {
+    const handleClick = (event: MouseEvent) => {
+      if (componentRef.current && !componentRef.current.contains(event.target as Node)) {
+        if (handleOnBlur) {
+          handleOnBlur();
+        }
+      }
+    };
+
     document.addEventListener('click', handleClick);
 
     return () => {
       document.removeEventListener('click', handleClick);
     };
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [handleOnBlur]);
```

Comment:
I found this useEffect is triggered every time you have a click, not sure if this the correct way to do it

Reply:
I tested that wrapped by useCallback could fix this
handleOnBlur={useCallback(() => onToggleOpen(false), [onToggleOpen])}

otherwise we are having unnecessary  re-attachments of the event listener